File: LeetCode/Easy/1475-final-prices-with-a-special-discount-in-a-shop/1475-final-prices-with-a-special-discount-in-a-shop.cpp
class Solution {
public:
    vector<int> finalPrices(vector<int>& prices) {
        int n = prices.size();
        vector<int> ret(n);

        for (int i = 0; i < n; ++i){
            int v = prices[i];
            for (int j = i + 1; j < n; ++j)
                if (prices[j] <= prices[i]){
                    v -= prices[j];
                    break;
                }
            ret[i] = v;
        }
        
        return ret;
    }
};

File: LeetCode/Easy/1475-final-prices-with-a-special-discount-in-a-shop/1475-final-prices-with-a-special-discount-in-a-shop.py
class Solution:
    def finalPrices(self, prices: List[int]) -> List[int]:
        n = len(prices)
        ret = [0] * n

        for i in range(n):
            v = prices[i]
            for j in range(i + 1, n):
                if prices[j] <= prices[i]:
                    v -= prices[j]
                    break
                
            ret[i] = v;
        
        
        return ret